rfcs/20230323-add-split-graph-op/README.md

```diff
@@ -0,0 +1,142 @@
+
+`Split` operation splits an input tensor into pieces of the specific length along some axis, which is opposite to the operation of `concat`. `Split` operation is supported by different deep learning frameworks and toolkit, eg. PyTorch [1], TensorFlow [2], OpenVINO [3], and ONNX [4].
+
+There is currently one Ali recommendation model from Alibaba customer, the goal is to speed up the model inference using ITEX and oneDNN Graph API by detecting and fusing the patterns defined in these libraries. Particularly, it is currently feasible to detect LayerNorm and MHA patterns in the model and generate efficient kernels for these patterns and obtain faster inference on Intel Xeon processors. Based on the analysis of Alibaba recommendation model, it was found that `Split` and `Concat` op (refer to the pic1) right before the MHA patterns occupy a large amount of time in the computation of the whole graph. Alibaba customer support team requests graph compiler backend to investigate possible optimizations for these patterns. Through analysis, it is found that the pattern of `split` + `concat` (refer to pic2) can be fused to improve performance. However, since the graph API does not support `Split` op, possible optimization is limited. So we propose to add the `Split` op into graph API.
+
+
+<div align=center><img src=images/graph-compute-time.png></div>
+<div align=center>pic1 compute time</div>
+
+<div align=center><img src=images/split-concat.png></div>
+<div align=center>pic2 split-concat</div>
+
+
+## Proposal
+
+The proposal is to add `Split` operation to the operation set of Graph API.
+
+### Operations
+
+`Split` operation splits one single input tensor (`data`) and generates multiple outputs (`output_i`) of the specific length along a scalar `axis` attribute. It produces multiple output tensors based on `num_splits` or `size_splits` attribute. The input and output tensors should have the same data type and support floating point data types (f32, bf16, and f16).
+
+
+#### Proposed Attributes
+
+`Split` operation supports three attributes: `axis`, `num_splits` and `size_splits`. Both `axis` and `num_splits` should be an interger value. `size_splits` is a 1-D Tensor (or list). The input data is split into `num_splits` or len(`size_splits`) elements. The shape of the i-th element has the same size as the value except along dimension axis where the size is determined by `num_splits` or `size_splits` attribute.
+
+Attribute Name | Description | Value Type |Supported Values | Required or Optional
+-- | -- | --| --|--
+`axis` | specifies which dimension to split along. |s64 |A s64 value in the range of [-r, r-1] where r = rank(data)  | Required
+`num_splits` | number of outputs into which the input tensor data will be split along axis dimension. |s64 |A s64 value in the range of[1, data.dims[axis]]  | Optional
+`size_splits` |  list of sizes for each output alone axis dims. |s64 |A s64 list containing positive values, none is default  | Optional
```

Comment:
Should it be `s64[]` instead of `s64` for the `Value Type` column?

Reply:
The value type here refers to the specific element type, which is consistent with other op definitions. The supported values column has a supplementary statement that it is a s64 list.

Reply:
> The `value type` here refers to the specific element type, which is consistent with other op definitions. The `supported values` column has a supplementary statement that it is a`s64`list.

I see